lightning/src/ln/payment_tests.rs

```diff
@@ -1371,3 +1379,188 @@ fn test_holding_cell_inflight_htlcs() {
+	// payment or (b) fail the payment.
+	do_test_intercepted_payment(InterceptTest::Forward);
+	do_test_intercepted_payment(InterceptTest::Fail);
+	// Make sure that intercepted payments will be automatically failed back if too many blocks pass.
+	do_test_intercepted_payment(InterceptTest::Timeout);
+}
+
+fn do_test_intercepted_payment(test: InterceptTest) {
+	let chanmon_cfgs = create_chanmon_cfgs(3);
+	let node_cfgs = create_node_cfgs(3, &chanmon_cfgs);
+
+	let mut zero_conf_chan_config = test_default_channel_config();
+	zero_conf_chan_config.manually_accept_inbound_channels = true;
+	let mut intercept_forwards_config = test_default_channel_config();
+	intercept_forwards_config.accept_intercept_htlcs = true;
+	let node_chanmgrs = create_node_chanmgrs(3, &node_cfgs, &[None, Some(intercept_forwards_config), Some(zero_conf_chan_config)]);
+
+	let nodes = create_network(3, &node_cfgs, &node_chanmgrs);
+	let scorer = test_utils::TestScorer::with_penalty(0);
+	let random_seed_bytes = chanmon_cfgs[0].keys_manager.get_secure_random_bytes();
+
+	let _ = create_announced_chan_between_nodes(&nodes, 0, 1, channelmanager::provided_init_features(), channelmanager::provided_init_features()).2;
+
+	let amt_msat = 100_000;
+	let intercept_scid = nodes[1].node.get_intercept_scid();
+	let payment_params = PaymentParameters::from_node_id(nodes[2].node.get_our_node_id())
+		.with_route_hints(vec![
+			RouteHint(vec![RouteHintHop {
+				src_node_id: nodes[1].node.get_our_node_id(),
+				short_channel_id: intercept_scid,
+				fees: RoutingFees {
+					base_msat: 1000,
+					proportional_millionths: 0,
+				},
+				cltv_expiry_delta: MIN_CLTV_EXPIRY_DELTA,
+				htlc_minimum_msat: None,
+				htlc_maximum_msat: None,
+			}])
+		])
+		.with_features(channelmanager::provided_invoice_features());
+	let route_params = RouteParameters {
+		payment_params,
+		final_value_msat: amt_msat,
+		final_cltv_expiry_delta: TEST_FINAL_CLTV,
+	};
+	let route = find_route(
+		&nodes[0].node.get_our_node_id(), &route_params, &nodes[0].network_graph, None, nodes[0].logger,
+		&scorer, &random_seed_bytes
+	).unwrap();
+
+	let (payment_hash, payment_secret) = nodes[2].node.create_inbound_payment(Some(amt_msat), 60 * 60).unwrap();
+	nodes[0].node.send_payment(&route, payment_hash, &Some(payment_secret), PaymentId(payment_hash.0)).unwrap();
+	let payment_event = {
+		{
+			let mut added_monitors = nodes[0].chain_monitor.added_monitors.lock().unwrap();
+			assert_eq!(added_monitors.len(), 1);
+			added_monitors.clear();
+		}
+		let mut events = nodes[0].node.get_and_clear_pending_msg_events();
+		assert_eq!(events.len(), 1);
+		SendEvent::from_event(events.remove(0))
+	};
+	nodes[1].node.handle_update_add_htlc(&nodes[0].node.get_our_node_id(), &payment_event.msgs[0]);
+	commitment_signed_dance!(nodes[1], nodes[0], &payment_event.commitment_msg, false, true);
+
+	// Check that we generate the PaymentIntercepted event when an intercept forward is detected.
```

Comment:
Would be valuable to also have test coverage ensuring that the `PaymentIntercepted` event isn't generated + `pending_intercepted_htlcs` isn't populated when `accept_intercept_htlcs` is false.

Reply:
Gonna save that for follow-up for the sake of the size of this PR, but you can manually check that the test fails when the config change lines are commented out. FWIW, we don't usually have that level of testing rigor with config options, i.e. the [zero-conf PR](https://github.com/lightningdevkit/rust-lightning/pull/1401) didn't/doesn't have it.

Reply:
Ok fair enough :)